src/nunit.analyzers.tests/DiagnosticSuppressors/DereferencePossiblyNullReferenceSuppressorTests.cs

```diff
@@ -638,9 +612,9 @@ public void HasCountShouldNotAffectNullabilitySuppression(List<int>? maybeNull)
                 }
             ", @"using System.Collections.Generic;");
 
-            await DiagnosticsSuppressorAnalyzer.EnsureSuppressed(suppressor,
-                DereferencePossiblyNullReferenceSuppressor.SuppressionDescriptors["CS8604"], testCode)
-                .ConfigureAwait(false);
+            RoslynAssert.Suppressed(suppressor,
+                ExpectedDiagnostic.Create(DereferencePossiblyNullReferenceSuppressor.SuppressionDescriptors["CS8602"]),
```

Comment:
Just curious. In moving the testing class to Gu.Roslyn.Assert was it improved so this noticed the wrong descriptor or did something else change ?

Reply:
Yes, it previously checked that the errors were suppressible, not if it was matching the id.
In the final test it only checked that the error was suppressed by this suppressor with the Justification.
The Justification (Assert.NotNull, etc) is the same regardless of CS error.